### PR TITLE
ARGO-395 Fix in response body in case wrong api-key is being used

### PR DESCRIPTION
**Pull request by pkoro**
# Description

Fix in response body in case a wrong api key is used under the /factors resource. Also updated unit tests.

Includes also a minor fix in swagger spec file.

### CODE REVIEW

**Comment by pkoro** [comment]
please review /cc @kaggis

File: doc/swagger/swagger.yaml [hunk 1]
```diff
@@ -1677,7 +1677,7 @@ definitions:
   FactorsResponse:
     type: object
     properties:
-      root:
+      factors:
```
Comment (kaggis) [inline doc/swagger/swagger.yaml]: Maybe a description should be added about what the resource: factors actually represents,  including the meaning of the fields "site" and "weight" and how they are relevant to computations/results

Comment (pkoro) [inline doc/swagger/swagger.yaml]: Descriptions in factors definition added in `swagger.yaml`

**Comment by themiszamani** [comment]
test this please

**Comment by themiszamani** [comment]
test this please

**Comment by themiszamani** [comment]
test this please

**Comment by themiszamani** [comment]
test this please

**Comment by themiszamani** [comment]
test this please

**Comment by themiszamani** [comment]
test this please

**Comment by themiszamani** [comment]
test this please

**Comment by themiszamani** [comment]
test this please

**Comment by themiszamani** [comment]
test this please

**Comment by themiszamani** [comment]
test this please

**Comment by themiszamani** [comment]
test this please

**Comment by themiszamani** [comment]
test this please

**Comment by kaggis** [comment]
+1

**Pull request merged**